File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/UserProfileRepo.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.ProfileRoles;
import io.curiousoft.ijudi.ordermanagement.model.UserProfile;

import java.util.List;
import java.util.Optional;

public interface UserProfileRepo extends ProfileRepo<UserProfile> {

    Optional<UserProfile> findByMobileNumber(String phone);

    List<UserProfile> findByRoleAndLatitudeBetweenAndLongitudeBetween(ProfileRoles messenger, double latMin, double latMax, double longMin, double longMax);
}

File: izinga-commons/src/main/kotlin/commons/repo/UserProfileRepo.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.ProfileRoles
import io.curiousoft.izinga.commons.model.UserProfile
import java.util.*

interface UserProfileRepo : ProfileRepo<UserProfile> {
    fun findByMobileNumber(phone: String?): Optional<UserProfile>?
    fun findByRoleAndLatitudeBetweenAndLongitudeBetween(
        messenger: ProfileRoles,
        latMin: Double,
        latMax: Double,
        longMin: Double,
        longMax: Double
    ): List<UserProfile>?
}